use of deprecated `bind2nd`

Build warnings under clang11.
I'm rather surprised that there *isn't* a deprecation warning in CI.

```
--- stderr: rttest                                                                           
/opt/ros/master/src/ros2/realtime_support/rttest/src/rttest.cpp:843:10: warning: 'bind2nd<std::__1::minus<int>, double>' is deprecated [-Wdeprecated-declarations]
    std::bind2nd(std::minus<int>(), output->mean_latency));
         ^
/usr/lib/llvm-11/bin/../include/c++/v1/functional:1089:1: note: 'bind2nd<std::__1::minus<int>, double>' has been explicitly marked deprecated here
_LIBCPP_DEPRECATED_IN_CXX11 inline _LIBCPP_INLINE_VISIBILITY
^
/usr/lib/llvm-11/bin/../include/c++/v1/__config:985:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#  define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                      ^
/usr/lib/llvm-11/bin/../include/c++/v1/__config:974:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
                                               ^
1 warning generated.
---
Finished <<< rttest [23.7s]
```

--- a/rttest/src/rttest.cpp
+++ b/rttest/src/rttest.cpp
@@ -840,7 +840,7 @@ int Rttest::calculate_statistics(struct rttest_results * output)
   std::vector<int64_t> latency_diff(latency_dataset.size());
   std::transform(
     latency_dataset.begin(), latency_dataset.end(), latency_diff.begin(),
-    std::bind2nd(std::minus<int>(), output->mean_latency));
+    std::bind(std::minus<int>(), std::placeholders::_1, output->mean_latency));
   int64_t sq_sum = std::inner_product(
     latency_diff.begin(), latency_diff.end(), latency_diff.begin(), 0);
   output->latency_stddev = std::sqrt(sq_sum / latency_dataset.size());

